src/h3/mod.rs

```diff
@@ -715,6 +715,10 @@ impl Connection {
         )?;
         conn.stream_send(stream_id, &header_block, fin)?;
 
+        if fin && conn.stream_finished(stream_id) {
```

Comment:
I think this is fine for normal operation, but makes me wonder how we handle stream collection when we do an early response. In that case we would trigger STOP_SENDING with H3_EARLY_RESPONSE, so its probably handled in code located elsewhere.

Reply:
That's what the TODO later is supposed to handle, among other things :)